### PR TITLE
DOC Fix documentation and README

### PR DESCRIPTION
**Pull request by janden**
Various fixes: backend discussions, benchmarks, source installation instructions, and smaller stuff.

### CODE REVIEW

**Comment by janden** [comment]
Closes #269, #270, #273, #279, and #283.

File: README.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
-Kymatio: wavelet scattering in PyTorch
+Kymatio: Wavelet scattering in PyTorch
 ======================================
 
 Kymatio is an implementation of the wavelet scattering transform in the Python programming language, suitable for large-scale numerical experiments in signal processing and machine learning.
+Scattering transforms are translation-invariant signal representations implemented as convolutional networks whose filters are not learned but fixed wavelet filters.
```
Comment (eickenberg) [inline README.md]: hmm does the last part read right to you? "are not [adjective] but [noun-related-thing]". I'd go with "not learned, but fixed (as wavelet filters)" or something

File: README.md [hunk 2]
```diff
@@ -50,20 +55,25 @@ Kymatio requires:
...
+```
+
+Once PyTorch is installed, you may install the latest version of Kymatio using the package manager `pip`, which will automatically download Kymatio from the Python Package Index (PyPI):
 
 ```
 pip install kymatio
 ```
 
-Linux and macOS are the two operating systems that are officially supported by Kymatio.
+Linux and macOS are the two officially supported operating systems.
```
Comment (eickenberg) [inline README.md]: MacOs?

File: doc/source/userguide.rst [hunk 3]
```diff
@@ -202,26 +185,21 @@ Currently, two backends exist for the 2D scattering transform:
...
 
-1-D backend
+1D backend
 -----------
 
 We compared our implementation with that of the ScatNet MATLAB package
 :cite:`anden2014scatnet` with similar settings. The following table shows the
-average computation time for a batch of size :math:`64 \times 1 \times 65536`:
+average computation time for a batch of size :math:`64 \times 65536`. This
+corresponds to :math:`64` signals containing :math:`65536`, or a total of about
```
Comment (eickenberg) [inline doc/source/userguide.rst]: Did we have dollar-math now or not?

File: README.md (hunk 1 above)
Comment (janden) [reply README.md]: You're right. I'll change it.

File: README.md (hunk 2 above)
Comment (janden) [reply README.md]: From what I understand, the approved capitalization is "macOS" (see https://en.wikipedia.org/wiki/MacOS#macOS).

File: doc/source/userguide.rst (hunk 3 above)
Comment (janden) [reply doc/source/userguide.rst]: I guess we do. I was too lazy to check. Will change.

**Pull request merged**